baseplate_test.go

```diff
@@ -229,7 +230,24 @@ type serviceConfig struct {
 	} `yaml:"redis"`
 }
 
-func TestDecodeConfigYAML(t *testing.T) {
+func writeToFile(t *testing.T, path, content string) {
+	t.Helper()
+
+	f, err := os.Create(path)
+	if err != nil {
+		t.Fatalf("Failed to create tmp config file %q: %v", path, err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			t.Fatalf("Failed to close tmp config file %q: %v", path, err)
+		}
+	}()
+	if _, err := io.Copy(f, strings.NewReader(content)); err != nil {
+		t.Fatalf("Failed to write to tmp config file %q: %v", path, err)
+	}
+}
```

Comment:
1. use os.WriteFile
2. do the tempdir call inside the helper (it'll be unique for each call) and return the path

```suggestion
func tempConfig(t *testing.T, content string) string {
	t.Helper()
  path := filepath.Join(t.TempDir(), "config.yaml"))
	if _, err := os.WriteFile(path, []byte(content)); err != nil {
		t.Fatalf("SETUP: Failed to write to tmp config file %q: %v", path, err)
	}
	return path
}
```

Reply:
Changed to `os.WriteFile`, and since that we no longer to wrap another helper function for the tests.